File: AoC2017Tests/Day01Test.fs
module Day01Test

open AoC2017
open NUnit.Framework
open System.Collections.Generic

[<TestCase("1122","3")>] 
[<TestCase("1111","4")>] 
[<TestCase("1234","0")>] 
[<TestCase("91212129","9")>]
let Part1 (input: string, expected : string) =
    let list = new List<string>();
    list.Add(input)
    let day = new Day01(list)
    let actual = day.SolvePart1()
    Assert.AreEqual(expected, actual)

[<Test>]
let Part2 () = 
    let day = new Day01(new List<string>())
    let actual = day.SolvePart2()
    Assert.AreEqual("", actual)

[<Test>]
let Part1Solution() = 
    let day = new Day01("day01")
    let actual = day.SolvePart1()
    Assert.AreEqual("1182", actual)

File: AoC2017Tests/Day01Test.cs
namespace AoC2017Tests;

public class Day01Test
{
	[TestCase("1122", "3")]
	[TestCase("1111", "4")]
	[TestCase("1234", "0")]
	[TestCase("91212129", "9")]
	public async Task Part1(string input, string expected)
	{
		var day01 = new Day01(new List<string> { input });
		var actual = await day01.SolvePart1( );
		Assert.That(actual, Is.EqualTo(expected));
	}

	[TestCase("1212", "6")]
	[TestCase("1221", "0")]
	[TestCase("123425", "4")]
	[TestCase("123123", "12")]
	[TestCase("12131415", "4")]
	public async Task Part2(string input, string expected)
	{
		var day01 = new Day01(new List<string> { input });
		var actual = await day01.SolvePart2( );
		Assert.That(actual, Is.EqualTo(expected));
	}
}
